client/components/community-translator/translatable.jsx

```diff
@@ -0,0 +1,258 @@
+ */
+import Dialog from 'components/dialog';
+import Button from 'components/button';
+import TranslatableTextarea from './translatable-textarea';
+import TranslatedSuccess from './translated-success';
+import { getTranslationData, getTranslationPermaLink, submitTranslation } from './utils.js';
+
+/**
+ * Module varialbles
+ */
+const TRANSLATION_FETCH_ERROR = 'translation-fetch-error';
+const TRANSLATION_SUBMIT_ERROR = 'tranlsation-submit-error';
+
+export class Translatable extends Component {
+	state = {
+		showDialog: false,
+		originalData: {},
+		formState: {},
+	};
+
+	hasDataLoaded() {
+		return ! isEmpty( this.state.originalData ) || ! isEmpty( this.state.error );
+	}
+
+	handleTranslationChange = event => {
+		const { name, value } = event.target;
+		this.setState( {
+			formState: {
+				...this.state.formState,
+				[ name ]: value,
+			},
+			submitting: false,
+			submissionSuccess: false,
+		} );
+	};
+
+	closeDialog = () => {
+		this.setState( {
+			showDialog: false,
+			submissionSuccess: false,
+			formState: this.state.submissionSuccess ? {} : this.state.formState,
+			originalData: this.state.submissionSuccess ? {} : this.state.originalData,
+		} );
+	};
+
+	openDialog = event => {
+		event.preventDefault();
+
+		this.setState( { showDialog: true } );
+
+		const { singular, context, plural, locale } = this.props;
+
+		if ( ! this.hasDataLoaded() ) {
+			getTranslationData( locale, { singular, context, plural } )
+				.then( originalData =>
+					this.setState( {
+						originalData,
+						translationUrl: getTranslationPermaLink( originalData.originalId, locale ),
+						formState: {
+							translatedSingular: originalData.translatedSingular,
+							translatedPlural: originalData.translatedPlural,
+						},
+					} )
+				)
+				.catch( () => {
+					this.setState( {
+						error: TRANSLATION_FETCH_ERROR,
+					} );
+				} );
+		}
+	};
+
+	submitForm = () => {
+		this.setState( {
+			submitting: true,
+		} );
+		submitTranslation( this.state.originalData.originalId, this.state.formState, this.props.locale )
+			.then( originalData => {
+				this.setState( {
+					error: originalData.error,
+					originalData,
+					submitting: false,
+					submissionSuccess: true,
+				} );
+			} )
+			.catch( () => {
+				this.setState( {
+					error: TRANSLATION_SUBMIT_ERROR,
+				} );
+			} );
+	};
+
+	getDialogButtons = () => {
+		const { translate } = this.props;
+
+		const buttons = [
+			<Button primary onClick={ this.closeDialog }>
+				{ translate( 'Close', { textOnly: true } ) }
+			</Button>,
+		];
+
+		if ( ! this.state.submissionSuccess ) {
+			buttons.push(
+				<Button
+					primary
+					onClick={ this.submitForm }
+					disabled={
+						this.state.originalData.translatedSingular ===
+							this.state.formState.translatedSingular &&
+						this.state.originalData.translatedPlural === this.state.formState.translatedPlural &&
+						! this.state.submitting
+					}
+				>
+					{ translate( 'Submit a new translation' ) }
```

Comment:
Hi! I've found a possible matching string that has already been translated **13** times:
`translate( 'Submit translation' )` *ES Score: 11*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).